Library/Homebrew/test/cask/cli/search_spec.rb

```diff
@@ -22,16 +22,38 @@
     EOS
   end
 
+  it "returns matches even when online search failed" do
+    allow(GitHub).to receive(:search_code).and_raise(GitHub::Error.new("reason"))
+    expect {
+      Hbc::CLI::Search.run("local")
+    }.to output(<<-EOS.undent).to_stdout
+      local-caffeine
+      local-transmission
+    EOS
+    .and output(/^Warning: Online search failed: reason/).to_stderr
+  end
+
   it "shows that there are no Casks matching a search term that did not result in anything" do
     expect {
       Hbc::CLI::Search.run("foo-bar-baz")
-    }.to output("No Cask found for \"foo-bar-baz\".\n").to_stdout.as_tty
+    }.to output(<<-EOS.undent).to_stdout.as_tty
+      No Cask found for "foo-bar-baz".
+    EOS
+  end
+
+  it "doesn't output anything to non-TTY stdout when there are no matches" do
+    expect {
+      Hbc::CLI::Search.run("foo-bar-baz")
+    }.to not_to_output.to_stdout
```

Comment:
`.not_to output.to_stdout`

Reply:
I tried to change it to `.not_to output.to_stdout.or output.to_stderr`, but then got the following error:

> `expect(...).not_to matcher.or matcher` is not supported, since it creates a bit of an ambiguity. Instead, define negated versions of whatever matchers you wish to negate with `RSpec::Matchers.define_negated_matcher` and use `expect(...).to matcher.or matcher`.

There are two matchers there: no output to stdout and no output to stderr. How should I rewrite it?

Reply:
Ok, then make it `.to not_output.to_stderr`, and add another negated matcher like `not_to_output`.

Reply:
Sorry, I'm not sure what you mean. It seems that it is already as you say, two negated conditions: `¬A & ¬B`:

```ruby
.to not_to_output.to_stdout
.and not_to_output.to_stderr
```

I changed the formatting in 8e5ad9a though.